File: codegen/nim/fixtures/resource/client/Users_service.nim
import marshal, tables
import client_struct

import City
import address
import usersGetRespBody
import usersuserIdaddressPostReqBody
import usersuserIdaddressPostRespBody


type
  Users_service* = object
    client*: Client
    name*: string

proc UsersSrv*(c : Client) : Users_service  =
  return Users_service(client:c, name:c.baseURI)


proc getUsers*(srv: Users_service, queryParams: Table[string, string] = initTable[string, string]()) : usersGetRespBody =
  # get users.# This method will be return list user.# Use it wisely.
  # It calls GET /users endpoint.

  let resp = srv.client.request("/users", "GET", queryParams=queryParams)
  return to[usersGetRespBody](resp.body)

proc usersPost*(srv: Users_service, reqBody: City, queryParams: Table[string, string] = initTable[string, string]()) : City =
  # create users
  # It calls POST /users endpoint.

  let resp = srv.client.request("/users", "POST", $$reqBody, queryParams=queryParams)
  return to[City](resp.body)

proc usersPut*(srv: Users_service, reqBody: seq[string], queryParams: Table[string, string] = initTable[string, string]()) : City =
  # create users
  # It calls PUT /users endpoint.

  let resp = srv.client.request("/users", "PUT", $$reqBody, queryParams=queryParams)
  return to[City](resp.body)

proc OptionsUsers*(srv: Users_service, queryParams: Table[string, string] = initTable[string, string]()) : string =
  
  # It calls OPTIONS /users endpoint.

  let resp = srv.client.request("/users", "OPTIONS", queryParams=queryParams)
  return to[string](resp.body)

proc GetUserByID*(srv: Users_service, userId: string, queryParams: Table[string, string] = initTable[string, string]()) : City =
  # get id
  # It calls GET /users/{userId} endpoint.

  let resp = srv.client.request("/users/"&userId, "GET", queryParams=queryParams)
  return to[City](resp.body)

proc usersByUserIdDelete*(srv: Users_service, userId: string, queryParams: Table[string, string] = initTable[string, string]()) : string =
  
  # It calls DELETE /users/{userId} endpoint.

  let resp = srv.client.request("/users/"&userId, "DELETE", queryParams=queryParams)
  return to[string](resp.body)

proc usersByUserIdAddressPost*(srv: Users_service, reqBody: usersuserIdaddressPostReqBody, userId: string, queryParams: Table[string, string] = initTable[string, string]()) : usersuserIdaddressPostRespBody =
  
  # It calls POST /users/{userId}/address endpoint.

  let resp = srv.client.request("/users/"&userId&"/address", "POST", $$reqBody, queryParams=queryParams)
  return to[usersuserIdaddressPostRespBody](resp.body)

proc usersByUserIdAddressFolderAddressIdTestAddressId2Get*(srv: Users_service, addressId: string, addressId2: string, userId: string, queryParams: Table[string, string] = initTable[string, string]()) : seq[address] =
  # get address id
  # It calls GET /users/{userId}/address/folder{addressId}test{addressId2} endpoint.

  let resp = srv.client.request("/users/"&userId&"/address/folder"&addressId&"test"&addressId2, "GET", queryParams=queryParams)
  return to[seq[address]](resp.body)

File: codegen/python/fixtures/client/aiohttp_client/users_service.py
class UsersService:
    def __init__(self, client):
        self.client = client



    async def get_users(self, headers=None, query_params=None, content_type="application/json"):
        """
        First line of comment.
        Second line of comment
        It is method for GET /users
        """
        uri = self.client.base_url + "/users"
        return await self.client.get(uri, headers, query_params, content_type)


    async def create_users(self, data, headers=None, query_params=None, content_type="application/json"):
        """
        create users
        It is method for POST /users
        """
        uri = self.client.base_url + "/users"
        return await self.client.post(uri, data, headers, query_params, content_type)


    async def option_users(self, headers=None, query_params=None, content_type="application/json"):
        """
        It is method for OPTIONS /users
        """
        uri = self.client.base_url + "/users"
        return await self.client.options(uri, headers, query_params, content_type)


    async def getuserid(self, userId, headers=None, query_params=None, content_type="application/json"):
        """
        get id
        It is method for GET /users/{userId}
        """
        uri = self.client.base_url + "/users/"+userId
        return await self.client.get(uri, headers, query_params, content_type)


    async def users_byUserId_delete(self, userId, headers=None, query_params=None, content_type="application/json"):
        """
        It is method for DELETE /users/{userId}
        """
        uri = self.client.base_url + "/users/"+userId
        return await self.client.delete(uri, headers, query_params, content_type)


    async def users_byUserId_address_byAddressId_get(self, addressId, userId, headers=None, query_params=None, content_type="application/json"):
        """
        get address id
        of address
        It is method for GET /users/{userId}/address/{addressId}
        """
        uri = self.client.base_url + "/users/"+userId+"/address/"+addressId
        return await self.client.get(uri, headers, query_params, content_type)
